feat: add configuration document (#1717)
vaadin/docs: articles/tools/observability/advanced/configuration.asciidoc

---
title: Configuration
order: 4
layout: page
---

= Configuration

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Author: I must have missed the Kit decision. But then it should still probably be `Vaadin Observability Kit`

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Reviewer: This needs more context IMO, people have no idea what a "default" vaadin instrumentation is.

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Reviewer 2: I checked this with Marc in the context of the documentation and he confirmed that we should use "Observability Kit".

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Reviewer: Maybe it would be better to move this part somewhere further down. This can create confusion, and it's less relevant than other parts of the configuration.

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Author: Removed all mentions of Vaadin and Observability

> The Vaadin Observability distribution by default sets the service name as `vaadin` and disables default vaadin- and servlet instrumentation.

Reviewer: We can still decide on the naming, and then give everything a final pass.

> The default instrumentation is disabled as a Vaadin application, that is a single page application, sends different requests for the same page making the defaults less helpful.

Reviewer: The explanation doesn't quite match my understanding of why we disabled it. The reason was more around the fact that logging all requests was not helpful / generates a lot of data, and that we wanted to have control over which requests generate a trace at all.

> The default instrumentation is disabled as a Vaadin application, that is a single page application, sends different requests for the same page making the defaults less helpful.

Author: updated

== Instrumentation Configuration

> Instrumentation can easily be configured in the `agent.properties` file that is used with the JavaAgent.

Reviewer: Subjective adjectives such as "easily" should be avoided IMO.

=== Service Configuration

> `service` attributes have special handling and are used to differentiate traces form different servers that are still for the same service that exist at the same time (e.g. horizontally scaled services).

Reviewer: ```suggestion
`service` attributes are used to differentiate traces from different services / applications which might have multiple instances running at the same time (e.g. horizontally scaled services).
```

Service attributes are as follows:

* `service.name` is an attribute that is used to distinguish a service by name.
+

> For Vaadin Observability this defaults to `vaadin`

Reviewer 2: ```suggestion
For Observability Kit, this defaults to `vaadin`
```

Author: I must have missed the Kit decision. But then it should still probably be `Vaadin Observability Kit`

Reviewer: `Vaadin Observability Kit agent` is overkill IMO, I don't think adding `Kit` to every mention of the agent is helpful or necessary.

Author: Changed to `custom distribution` instead of Observability

* `service.namespace` helps to distinguish a group of services

> * `service.instance.id` helps to distinguish instances of the same service that exist at the same time

Reviewer: Technically there are also [`host`](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/host/) attributes for this purpose, Marc mentioned that these are used by some vendors.

Author: I can find host settings are only at experimental status
```
Host
Status: Experimental
```

Reviewer: Alright, let's skip that for now.

+
Must be unique for each instance of the same `service.namespace`, `service.name` pair
* `service.version` semver version string of the service

Only `service.name` has a default value as it is the only one required.

For detailed information see https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/#service[Service]

=== Disabled Default Instrumentation

The by default disabled instrumentation modules are:

* java-http-client
* jetty
* jetty-httpclient
* servlet
* tomcat

> To enable any of these, add to the `agent.properties` file the line `otel.instrumentation.${instrumentationName}.enabled=true`

Reviewer 2: ```suggestion
To enable any of these, add this line to the `agent.properties` file:
[source,properties]
----
`otel.instrumentation.${instrumentationName}.enabled=true`
----
```

---
title: Configuration
order: 4
layout: page
---

= Configuration

> == Instrumentation Configuration

Author: dropped
